src/main/java/seedu/address/storage/volunteer/JsonAdaptedVolunteer.java

```diff
@@ -28,26 +33,43 @@
  */
 public class JsonAdaptedVolunteer extends JsonAdaptedPerson implements JsonSerializable<Volunteer> {
     public static final String MISSING_FIELD_MESSAGE_FORMAT = "Volunteer's %s field is missing!";
+    private final List<JsonAdaptedMedicalTag> medicalTags = new ArrayList<>();
 
     /**
      * Constructs a {@code JsonAdaptedVolunteer} with the given volunteer details.
      */
     @JsonCreator
     public JsonAdaptedVolunteer(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
-            @JsonProperty("email") String email, @JsonProperty("address") String address,
-            @JsonProperty("nric") String nric, @JsonProperty("age") String age,
-            @JsonProperty("region") String region,
-            @JsonProperty("tagged") List<JsonAdaptedTag> tagged,
-            @JsonProperty("availableDates") List<JsonAdaptedAvailableDate> dates) {
+                                @JsonProperty("email") String email, @JsonProperty("address") String address,
+                                @JsonProperty("nric") String nric, @JsonProperty("age") String age,
+                                @JsonProperty("region") String region,
+                                @JsonProperty("tagged") List<JsonAdaptedTag> tagged,
+                                @JsonProperty("medicalTagged") List<JsonAdaptedMedicalTag> medicalTagged,
+                                @JsonProperty("availableDates") List<JsonAdaptedAvailableDate> dates) {
```

Comment:
I think the code editor automatically indented this, perhaps we should keep it to 8 spaces

```suggestion
            @JsonProperty("email") String email, @JsonProperty("address") String address,
            @JsonProperty("nric") String nric, @JsonProperty("age") String age,
            @JsonProperty("region") String region,
            @JsonProperty("tagged") List<JsonAdaptedTag> tagged,
            @JsonProperty("medicalTagged") List<JsonAdaptedMedicalTag> medicalTagged,
            @JsonProperty("availableDates") List<JsonAdaptedAvailableDate> dates) {
```